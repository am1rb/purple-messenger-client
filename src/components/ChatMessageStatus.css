.status
{
	color: #5ca04a;
	display: inline-block;
	width: 2.4em;
	text-align: right;
	position: relative;
	font-size: 0.7em;
}
