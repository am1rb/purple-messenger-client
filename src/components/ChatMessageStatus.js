// @flow

import React from 'react'

import styles from './ChatMessageStatus.css'

type Props = {
	status: string,
};

export default class ChatMessageStatus extends React.PureComponent<Props>
{
	render()
	{
		return (
			<span className={styles.status}>
				{this.props.status==='pending' && <i className="far fa-clock"></i>}
				{this.props.status!=='pending' && <i className="fa fa-check"></i>}
				{this.props.status!=='pending' && this.props.status==='received' && <i className="fa fa-check"></i>}
			</span>
		)
	}
}