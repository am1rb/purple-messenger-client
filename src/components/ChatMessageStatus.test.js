// @flow

import React from 'react'
import renderer from 'react-test-renderer'

import ChatMessageStatus from './ChatMessageStatus'


describe("<ChatMessageStatus /> component tests", () => {

	it("Should matches to pending snapshot", () => {
		const component = renderer.create(
			<ChatMessageStatus status="pending" />
		)
		expect(component.toJSON()).toMatchSnapshot()
	})

	it("Should  matches to received snapshot", () => {
		const component = renderer.create(
			<ChatMessageStatus status="received" />
		)
		expect(component.toJSON()).toMatchSnapshot()
	})

})

