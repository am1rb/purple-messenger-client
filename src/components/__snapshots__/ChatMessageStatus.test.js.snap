// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessageStatus /> component tests Should  matches to received snapshot 1`] = `
<span
  className={undefined}
>
  <i
    className="fa fa-check"
  />
  <i
    className="fa fa-check"
  />
</span>
`;

exports[`<ChatMessageStatus /> component tests Should matches to pending snapshot 1`] = `
<span
  className={undefined}
>
  <i
    className="far fa-clock"
  />
</span>
`;
